From bfb0e8e4de740c07742c7df217b32fdbf2c92536 Mon Sep 17 00:00:00 2001
From: Igor Raits <i.gnatenko.brain@gmail.com>
Date: Mon, 15 Jun 2020 19:33:52 +0200
Subject: [PATCH 1/2] macros.meson: Switch to %{_smp_build_ncpus}

It is available since RPM 4.15 which has been around 1 year by now.

Signed-off-by: Igor Raits <i.gnatenko.brain@gmail.com>
---
 data/macros.meson | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/data/macros.meson b/data/macros.meson
index c5b90de03..25601aad3 100644
--- a/data/macros.meson
+++ b/data/macros.meson
@@ -2,12 +2,6 @@
 %__meson_wrap_mode nodownload
 %__meson_auto_features enabled
 
-%_smp_mesonflags %([ -z "$MESON_BUILD_NCPUS" ] \\\
-	&& MESON_BUILD_NCPUS="`/usr/bin/getconf _NPROCESSORS_ONLN`"; \\\
-        ncpus_max=%{?_smp_ncpus_max}; \\\
-        if [ -n "$ncpus_max" ] && [ "$ncpus_max" -gt 0 ] && [ "$MESON_BUILD_NCPUS" -gt "$ncpus_max" ]; then MESON_BUILD_NCPUS="$ncpus_max"; fi; \\\
-        if [ "$MESON_BUILD_NCPUS" -gt 1 ]; then echo "--num-processes $MESON_BUILD_NCPUS"; fi)
-
 %meson \
     %set_build_flags \
     %{shrink:%{__meson} \
@@ -37,8 +31,8 @@
     %ninja_install -C %{_vpath_builddir}
 
 %meson_test \
-    %{shrink: %{__meson} test \
+    %{shrink:%{__meson} test \
         -C %{_vpath_builddir} \
-        %{?_smp_mesonflags} \
+        --num-processes %{_smp_build_ncpus} \
         --print-errorlogs \
-    %{nil}}
+        %{nil}}
-- 
2.27.0

